compiler/codegen/OMRCodeGenerator.cpp

```diff
@@ -3525,7 +3659,17 @@ OMR::CodeGenerator::isMaterialized(TR::Node * node)
    }
 
 
-// determine if value fits in the immediate field (if any) of instructions that machine provides
+/**
+ * \brief
+ * Determine if value fits in the immediate field (if any) of instructions that machine provides.
+ *
+ * \details
+ *
+ * A node with large constant can be materialized and left as commoned nodes.
+ * Smaller constants can be uncommoned so that they re-materialize everytime when needed as a call
+ * parameter. This query is platform specific as constant loading can be expensive on some platforms
+ * but cheap on others, depending on the their magnitude.
+ */
 bool
 OMR::CodeGenerator::isMaterialized(int64_t value)
```

Comment:
How about sinking the following three functions to the `omr/compiler/<platform>/codegen` folders so that they are not exposed in the common code generator layer?

`isMaterialized(int64_t value)`
`SmallestPosConstThatMustBeMaterialized()`
`LargestNegConstThatMustBeMaterialized()`